packages/js/product-editor/src/contexts/validation-context/helpers.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { ValidationErrors } from './types';
+
+export function findFirstInvalidElement< E extends Element = Element >(
+	elementsMap: Record< string, E >,
+	errors: ValidationErrors
+): E | undefined {
+	const fieldRefsWithError = Object.entries( elementsMap ).filter(
+		( [ validatorId, element ] ) =>
+			// Pick the element if it is under the selected tab.
+			element?.closest( '.is-selected[role="tabpanel"]' ) &&
```

Comment:
Do we only want to find invalid elements under the currently selected tab?  I would think we just find the first and then change to that tab.

Also noting that this function is currently very tied to our UI, which we may consider decoupling a bit later, but I think this is a good first iteration.

Reply:
Good question. We may want to change tabs if necessary, but the more I think about it, the more sense it makes to use the pre-publish modal from the post editor and surface the errors there. Combining both solutions (focusing on the first affected field and having a clear list of all errors) may produce the best UX.

Reply:
I'm good with leaving this as is until we create a way to surface all errors easily.

>If we are going to change the selected tab that contains the invalid field then we need to know the structure of the UI which will make the implementation very coupled.

One possible solution to this if we need it is to provide a path option along with the field ref.  Since tab selection is also dictated by URL, we could switch to `...&tab=general` prior to focus which doesn't tie us specifically to our UI or require knowledge around UI elements.